pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -9,6 +9,28 @@ shopt -s inherit_errexit
 
 export PATH=@path@:$PATH
 
+
+# save output piped to msgPipe or 'msg message' to systemd journal
+# 'journalctl -t nixos-rebuild' to inspect
+# tee is used so we still get output to the terminal
+# check if systemd-cat exists and if systemctl exits with 0. if it does not
+# then systemd isn't running(chroot) and we cannot use systemd-cat.
+
+# msg something
+msg() {
+  if command -v systemd-cat &>/dev/null && systemctl &>/dev/null; then
```

Comment:
[`command -v` might be better as `type`](https://unix.stackexchange.com/q/85249/3645). Also, what is the purpose of the `systemctl` command here? If it's just to check whether that command exists, it's pretty slow. You might want to use `type systemctl`.